source/Ship.cpp

```diff
@@ -913,13 +945,18 @@ void Ship::Save(DataWriter &out) const
 					hardpoint.GetOutfit()->Name());
 			else
 				out.Write(type, 2. * hardpoint.GetPoint().X(), 2. * hardpoint.GetPoint().Y());
-			double hardpointAngle = hardpoint.GetBaseAngle().Degrees();
+			const auto &attributes = hardpoint.GetBaseAttributes();
+			const double baseDegree = attributes.baseAngle.Degrees();
+			const double firstArc = round((attributes.minArc.Degrees() - baseDegree)*100)/100;
+			const double secondArc = round((attributes.maxArc.Degrees() - baseDegree)*100)/100;
```

Comment:
You add an error that is less than ±0.005 degrees. The save-load cycle should be transparent. If you really want to round it, you should do it when loading instead of saving.
The LSB of Angle is 360/65536 ≒ 0.0054931640625 degrees; you have no need to care a difference less than LSB/2. The save file is not a user interface.